packages/connections/src/ConnectionEvent.js

```diff
@@ -0,0 +1,46 @@
+const OPEN = Symbol("OPEN");
+const DATA = Symbol("DATA");
+const ERROR = Symbol("ERROR");
+const COMPLETE = Symbol("COMPLETE");
+const ABORT = Symbol("ABORT");
+
+export default class ConnectionEvent {
+  /**
+   * Connection event
+   * @param {EventEmitter} target
+   * @param {Symbol} type
+   */
+  constructor(target, type) {
```

Comment:
Let's throw an error if these arguments aren't supplied. Otherwise we've created a type construct that still fails to guarantee structure.

Reply:
<img src="https://cloud.githubusercontent.com/assets/647035/21101001/3a24fe0a-c077-11e6-810d-95006b39fa85.gif" width="100%" />